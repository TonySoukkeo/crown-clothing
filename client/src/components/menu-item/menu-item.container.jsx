import { withRouter } from "react-router-dom";

import MenuItem from "./menu-item.component";

const MenuItemContainer = withRouter(MenuItem);

export default MenuItemContainer;
