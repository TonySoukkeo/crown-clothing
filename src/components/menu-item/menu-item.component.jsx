import React from "react";
import { withRouter } from "react-router-dom";

import {
  BackgroundImageContainer,
  MenuItemContainer,
  ContentContainer,
  TitleContainer,
  SubTitleContainer,
} from "./menu-item.styles.jsx";
// import "./menu-item.styles.scss";

const menuItem = ({ title, imageUrl, size, history, linkUrl, match }) => {
  return (
    <MenuItemContainer
      size={size}
      onClick={() => history.push(`${match.url}${linkUrl}`)}
    >
      <BackgroundImageContainer
        style={{
          backgroundImage: `url(${imageUrl})`,
        }}
      >
        <ContentContainer>
          <TitleContainer>{title.toUpperCase()}</TitleContainer>
          <SubTitleContainer>SHOP NOW</SubTitleContainer>
        </ContentContainer>
      </BackgroundImageContainer>
    </MenuItemContainer>
  );
};

export default withRouter(menuItem);
