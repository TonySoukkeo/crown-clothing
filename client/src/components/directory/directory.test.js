import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Directory from "./directory.component";
import { default as MenuItem } from "../menu-item/menu-item.container";

configure({
  adapter: new Adapter(),
});

describe("Directory component", () => {
  let wrapper;
  let mockSections;
  beforeEach(() => {
    mockSections = [{ id: 1, id: 2 }];

    wrapper = shallow(<Directory sections={mockSections} />);
  });

  it("renders Directory component without crashing", () => {
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("displays the correct number of MenuItems", () => {
    expect.assertions(1);
    expect(wrapper.find(MenuItem).length).toEqual(mockSections.length);
  });

  it("outputs an empty array if there are no sections", () => {
    const mockSections = [];

    const wrapper2 = shallow(<Directory sections={mockSections} />);

    expect(wrapper2.children().length).toEqual(0);
  });
});
