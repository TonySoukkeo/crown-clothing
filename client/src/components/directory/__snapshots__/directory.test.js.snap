// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component renders Directory component without crashing 1`] = `ShallowWrapper {}`;
